### PR TITLE
Add Bing Search functionality and update README

### PR DESCRIPTION
**Pull request by ForestLinSen**
### Background
This pull request aims to enhance the search capabilities of the application by adding Bing Search functionality as an alternative to Google Search. Users can now choose between Google and Bing as their preferred search engine by specifying the --bing flag when running the program. This change aligns with the project's goal of providing versatile and user-friendly tools for information retrieval.

### Changes
1. Implemented the bing_search function to support Bing Search API integration.
2. Modified the execute_command function to accept a search_engine parameter, allowing for seamless integration of additional search engines in the future.
3. Updated the README.md file with clear instructions on how to enable Bing Search and set up Azure API keys for Bing Search support.

### Documentation
In-code comments have been added to explain the new bing_search function and the modifications made to the execute_command function. The README.md file has been updated to provide detailed instructions on enabling Bing Search and setting up Azure API keys, making it easy for users to understand and utilize the new feature.

### Test Plan
Tested the Bing Search functionality with various prompts to ensure accurate search results.
Verified that the --bing flag works as intended, allowing users to switch between Google and Bing as their search engine.
Ensured that the application continues to work as expected when no search engine flag is specified (defaults to Google Search).

### PR Quality Checklist
- [x] My pull request is atomic and focuses on a single change.
- [x] I have thoroughly tested my changes with multiple different prompts.
- [x] I have considered potential risks and mitigations for my changes.
- [x] I have documented my changes clearly and comprehensively.
- [x] I have not snuck in any "extra" small tweaks changes

### CODE REVIEW

**Review by nponeccop — DISMISSED** [review]

File: scripts/main.py [hunk 1]
```diff
@@ -438,4 +443,4 @@ def start_interaction_loop(self):
 
 
 if __name__ == "__main__":
-    main()
+    main()
```
Comment (nponeccop) [inline scripts/main.py]: There is whitespace error here. Press Enter once after main(). It won't pass the linter anyway.

File: scripts/main.py [hunk 2]
```diff
@@ -16,7 +16,6 @@
 import argparse
 from logger import logger
 import logging
-from prompt import get_prompt
```
Comment (nponeccop) [inline scripts/main.py]: While you are at it, revert this change as well as it's unrelated to the rest of the PR. You can submit it as a separate PR. We are happy to merge one-liners.

**Review by nponeccop — CHANGES_REQUESTED** [review]

**Comment by Artemonim** [comment]
Why didn't you change and use `.env.template`?

**Comment by ForestLinSen** [comment]
> Why didn't you change and use `.env.template`?

Thank you for your suggestion! I have updated the .env.template file to include the AZURE_API_KEY variable

**Comment by Artemonim** [comment]
> > Why didn't you change and use `.env.template`?
> 
> Thank you for your suggestion! I have updated the .env.template file to include the AZURE_API_KEY variable

I've mean `SEARCH_ENGINE=your-choice`

**Review by nponeccop — APPROVED** [review]

**Comment by ForestLinSen** [comment]
@nponeccop Hi! just want to update that I have resolved the conflicts and broken code that were introduced by the module introduction

**Review by nponeccop — APPROVED** [review]

**Comment by nponeccop** [comment]
@ForestLinSen There are conflicts now

**Review by nponeccop — CHANGES_REQUESTED** [review]
There are conflicts

**Comment by ForestLinSen** [comment]
> There are conflicts

Hi @nponeccop , the conflicts are resolved now

**Review by nponeccop — APPROVED** [review]

**Review by nponeccop — APPROVED** [review]

**Comment by nponeccop** [comment]
@richbeales @p-i- Can you provide the author with some guidance as for the planned plugins feature?

**Comment by ForestLinSen** [comment]
Hi @nponeccop , I refactored the code as a plugin following this template: https://github.com/Significant-Gravitas/Auto-GPT-Plugin-Template. Should I modify and reopen the pull request here, create a new one, or follow any other guidance?

**Comment by ForestLinSen** [comment]
> Is this feature still engaging? Or is it abandoned?

Hi, this function has been integrated into the first-party plugins, you can find more information here: https://github.com/Significant-Gravitas/Auto-GPT-Plugins/tree/master

**Pull request closed without merging**